iml-agent/src/action_plugins/lustre/snapshot.rs

```diff
@@ -3,7 +3,31 @@
 // license that can be found in the LICENSE file.
 
 use crate::{agent_error::ImlAgentError, lustre::lctl};
-use iml_wire_types::snapshot::{Create, Destroy, Mount, Unmount};
+use combine::{stream::easy, EasyParser};
+use iml_wire_types::snapshot::{Create, Destroy, List, Mount, Snapshot, Unmount};
+
+mod parse;
+
+pub async fn list(l: List) -> Result<Vec<Snapshot>, ImlAgentError> {
+    let mut args = vec!["snapshot_list", "--fsname", &l.fsname];
```

Comment:
This could also be done immutably with something like:
```
let args: Vec<String> = vec![
        Some(vec!["snapshot_list".to_string(), "--fsname".to_string(), l.fsname]),
        l.name.map(move |name| vec!["--name".to_string(), name.to_string()]),
        match l.detail {
            true => Some(vec!["--detail".to_string()]),
            false => None,
        }
    ]
    .into_iter()
    .filter_map(|xs| xs)
    .flatten()
    .collect();
```